hyperspy/_components/offset.py

```diff
@@ -92,9 +95,19 @@ def estimate_parameters(self, signal, x1, x2, only_current=False):
             dc = signal.data
             gi = [slice(None), ] * len(dc.shape)
             gi[axis.index_in_array] = slice(i1, i2)
-            self.offset.map['values'][:] = dc[gi].mean(axis.index_in_array)
-            if binned is True:
+            self.offset.map['values'][:] = dc[tuple(gi)].mean(axis.index_in_array)
+            if self.binned:
                 self.offset.map['values'] /= axis.scale
             self.offset.map['is_set'][:] = True
             self.fetch_stored_values()
             return True
+
+    def function_nd(self, axis):
```

Comment:
I haven't rewrote this component to an `Expression`, because it looks to me that the expression string needs to have a free parameter, which is not the case here! Anyway, it seems unlikely (see documentation available online) that there will be a speed improvement using numexpr in this specific case.